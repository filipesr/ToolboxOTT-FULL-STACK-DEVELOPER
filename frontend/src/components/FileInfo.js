import { Table } from "react-bootstrap";

function FileInfo({ fileData }) {
  const { file, lines } = fileData;

  return (
    <Table striped bordered hover>
      <thead>
        <tr>
          <th>File Name</th>
          <th>Text</th>
          <th>Number</th>
          <th>Hex</th>
        </tr>
      </thead>
      <tbody>
        {lines.map((line, key) => (
          <tr key={key}>
            <td>{file}</td>
            <td>{line.text}</td>
            <td>{line.number}</td>
            <td>{line.hex}</td>
          </tr>
        ))}
      </tbody>
    </Table>
  );
}

export default FileInfo;
