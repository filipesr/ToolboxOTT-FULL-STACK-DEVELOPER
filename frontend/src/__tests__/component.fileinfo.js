import React from "react";
import FileInfo from "../components/FileInfo";
import { create } from "react-test-renderer";

describe("FileInfo coponent", () => {
  const fileData = {
    file: "teste",
    lines: [{ text: "a", number: 0, hex: "a" }],
  };
  test("testing reading data file", () => {
    let tree = create(<FileInfo fileData={fileData} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
