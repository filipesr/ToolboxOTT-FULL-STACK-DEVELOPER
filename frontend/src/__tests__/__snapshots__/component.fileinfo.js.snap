// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileInfo coponent testing reading data file 1`] = `
<table
  className="table table-striped table-bordered table-hover"
>
  <thead>
    <tr>
      <th>
        File Name
      </th>
      <th>
        Text
      </th>
      <th>
        Number
      </th>
      <th>
        Hex
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        teste
      </td>
      <td>
        a
      </td>
      <td>
        0
      </td>
      <td>
        a
      </td>
    </tr>
  </tbody>
</table>
`;
